go/libraries/doltcore/sqle/dolt_log_table_function.go

```diff
@@ -142,24 +163,78 @@ func (ltf *LogTableFunction) Expressions() []sql.Expression {
 	return exprs
 }
 
-// WithExpressions implements the sql.Expressioner interface.
-func (ltf *LogTableFunction) WithExpressions(expression ...sql.Expression) (sql.Node, error) {
-	if len(expression) > 2 {
-		return nil, sql.ErrInvalidArgumentNumber.New(ltf.FunctionName(), "0 to 2", len(expression))
+func (ltf *LogTableFunction) getDoltArgs(ctx *sql.Context, expressions []sql.Expression) ([]string, error) {
+	var args []string
+
+	for _, expr := range expressions {
+		childVal, err := expr.Eval(ctx, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		if !sql.IsText(expr.Type()) {
+			return args, sql.ErrInvalidArgumentDetails.New(ltf.FunctionName(), expr.String())
+		}
+
+		text, err := sql.Text.Convert(childVal)
+		if err != nil {
+			return nil, err
+		}
+
+		if text != nil {
+			args = append(args, text.(string))
+		}
 	}
 
+	return args, nil
+}
+
+// WithExpressions implements the sql.Expressioner interface.
+func (ltf *LogTableFunction) WithExpressions(expression ...sql.Expression) (sql.Node, error) {
 	for _, expr := range expression {
 		if !expr.Resolved() {
 			return nil, ErrInvalidNonLiteralArgument.New(ltf.FunctionName(), expr.String())
 		}
 	}
 
-	exLen := len(expression)
+	args, err := ltf.getDoltArgs(ltf.ctx, expression)
+	if err != nil {
+		return nil, err
+	}
+
+	apr, err := cli.CreateLogArgParser().Parse(args)
+	if err != nil {
+		return nil, sql.ErrInvalidArgumentDetails.New(ltf.FunctionName(), err.Error())
+	}
+
+	if notRevisionStr, ok := apr.GetValue(cli.NotFlag); ok {
+		ltf.notRevision = notRevisionStr
+	}
+
+	minParents := apr.GetIntOrDefault(cli.MinParentsFlag, 0)
+	if apr.Contains(cli.MergesFlag) {
+		minParents = 2
+	}
+	ltf.minParents = minParents
+
+	// Gets revisions, excluding any flag-related expression
+	var filteredExpressions []sql.Expression
+	for i, ex := range expression {
+		if !strings.Contains(ex.String(), "--") && !(i > 0 && strings.Contains(expression[i-1].String(), "--")) {
+			filteredExpressions = append(filteredExpressions, ex)
+		}
+	}
```

Comment:
Not sure if this is the best way to handle the expressions/flags. We use the arg parser to handle arguments for procedures, but that converts all the arguments to primitive types instead of expressions, which we seem to rely on more heavily in these table functions. Right now this is converting flags to a primitive types and keeps the revisions as sql expressions